src/ttt_core/console.clj

```diff
@@ -59,50 +65,73 @@
 (defn draw-main [game message]
   (do
     (clear-terminal)
-    (draw-header)
+    (draw-header (:language game))
     (draw-player-info game)
     (draw-board (:board game))
     (println message)
     (draw-footer)))
 
-(defn build-choose-game-mode-string 
-  ([] (build-choose-game-mode-string ""))
-  ([message]
-    (str message "Choose a game mode:\n1. Human-vs-Human\n2. Human-vs-Computer")))
-
-(defn request-game-mode
-  ([game] (request-game-mode game ""))
-  ([game message]
-    (let [modes {:1 :human-vs-human :2 :human-vs-computer}]
+(defn build-selection-string 
+  ([language msg-keyword] (build-selection-string language msg-keyword ""))
+  ([language msg-keyword prepend-msg]
+    (str prepend-msg
+      (if (= "" prepend-msg) "" " ")
+      (translate [language] [msg-keyword]))))
+        
+(defn request-game-setup
+  ([game] (request-game-setup game ""))
+  ([game prepend-msg]
+    (let [opts {:1 :human-vs-human :2 :human-vs-computer :3 :language-setup}]
+      (do
+        (draw-main game (translate [(:language game)] [:game-setup-selection-header]))
+        (println (build-selection-string (:language game) :choose-game-setup prepend-msg))
+        (get opts (keyword (read-line)))))))
+
+(defn request-language-setup
+  ([game] (request-language-setup game ""))
+  ([game prepend-msg]
+    (let [opts {:1 :en :2 :pl}]
       (do
-        (draw-main game "--- Game Mode Selection ---")
-        (println (build-choose-game-mode-string message))
-        (get modes (keyword (read-line)))))))
+        (draw-main game (translate [(:language game)] [:game-language-selection-header]))
+        (println (build-selection-string (:language game) :choose-language-setup prepend-msg))
+        (get opts (keyword (read-line)))))))
 
-(defn get-game-mode-selection [game] 
-  (loop [selection (request-game-mode game)]
+(defn get-selection [game request-fn] 
+  (loop [selection (request-fn game)]
     (if (not (nil? selection))
       selection
-      (recur (request-game-mode game "Invalid selection. ")))))
-
-(defn handle-game-mode-selection [game]
-  (assoc game :game-mode (get-game-mode-selection game)))
-
-(defn current-player-name [current-player-token game-mode]
-  (if (= game-mode :human-vs-human)
-    (if (= current-player-token :player-1-token) "Player 1" "Player 2")
-    (if (= current-player-token :player-1-token) "Player" "Computer")))
-
-(defn build-current-player-string [name]
-  (str name "'s move!"))
+      (recur (request-fn game (translate [(:language game)] [:invalid-selection]))))))
+
+(defn handle-language-selection [game]
+  (assoc game :language (get-selection game request-language-setup)))
+
+(defn handle-game-setup [game]
```

Comment:
`handle-game-setup` includes branching logic to call `handle-language-selection` if `request-game-setup` returns `:language-selection`, else the `selection` return is set to `:game-mode` and the game progresses. The reason for this somewhat entangled logic is that the language selection story should be optional, meaning that the user should not have to select English as the default language. The idea was to avoid entering a language selection prompt unless the user requests as much during the game mode setup.